bin/bot/src/discord/slash/reputation.rs

```diff
@@ -64,13 +124,68 @@ pub async fn run(ctx: &Context, command: &ApplicationCommandInteraction) -> sere
     if subcommand.kind == CommandOptionType::SubCommand {
         match subcommand.name.as_str() {
             "view" => view(ctx, command, &subcommand.options).await?,
+            "add" => update(ctx, command, &subcommand.options, true).await?,
+            "remove" => update(ctx, command, &subcommand.options, false).await?,
             // "event" => event(ctx, command, &subcommand.options).await?,
             _ => unreachable!(),
         }
     }
     Ok(())
 }
 
+async fn update(
+    ctx: &Context,
+    command: &ApplicationCommandInteraction,
+    options: &[CommandDataOption],
+    add: bool,
+) -> serenity::Result<()> {
+    let mut interaction = Interaction::new(ctx, Generic::Application(command), options);
+    super::requires_roles(
+        command.user.id,
+        &[STAFF],
+        &command
+            .member
+            .as_ref()
+            .expect("member should always exist on guild commands")
+            .roles,
+        ShouldAsk::Deny,
```

Comment:
im assuming this is what creates a button to allow or deny a command.
Should this be turned to allow?